src/genome.py

```diff
@@ -204,7 +289,46 @@ def copy_te(self, te: int, offset: int) -> int | None:
 
         If te is not active, return None (and do not copy it).
         """
-        ...  # FIXME
+
+        if te not in self.active_TE:
+            return None
+
+        pos = 0
+
+        currentNode = self.head
+        # find the first node and position of it, in the target TE.
+        for _ in range(self.length):
+            if currentNode.te_id == te:
+                break
+            currentNode = currentNode.next
+            pos += 1
+
+        TE_ID = self.next_TE_id
+        # path to node before insertion
+        pos = pos + (offset % self.length)
+        currentNode = self.head
+
+        for _ in range(0, pos - 1):
+            currentNode = currentNode.next
```

Comment:
If the offset is small, you can lose quite a bit of time by searching for the position from the start of the genome. With a doubly-linked list you could just move `offset` steps to either side.